runtime/runtime_controller_unittests.cc

```diff
@@ -0,0 +1,143 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/runtime/runtime_controller.h"
+#include "flutter/runtime/runtime_delegate.h"
+
+#include "flutter/lib/ui/semantics/semantics_update.h"
+#include "flutter/shell/common/shell_test.h"
+#include "flutter/testing/testing.h"
+
+namespace flutter {
+
+namespace testing {
+
+using RuntimeControllerTest = ShellTest;
+
+class MockRuntimeDelegate : public RuntimeDelegate {
+ public:
+  FontCollection font;
+  std::vector<SemanticsNodeUpdates> updates;
+  std::vector<CustomAccessibilityActionUpdates> actions;
+  std::string DefaultRouteName() override { return ""; }
+
+  void ScheduleFrame(bool regenerate_layer_trees = true) override {}
+
+  void OnAllViewsRendered() override {}
+
+  void Render(int64_t view_id,
+              std::unique_ptr<flutter::LayerTree> layer_tree,
+              float device_pixel_ratio) override {}
+
+  void UpdateSemantics(SemanticsNodeUpdates update,
+                       CustomAccessibilityActionUpdates actions) override {
+    this->updates.push_back(update);
+    this->actions.push_back(actions);
+  }
+
+  void HandlePlatformMessage(
+      std::unique_ptr<PlatformMessage> message) override {}
+
+  FontCollection& GetFontCollection() override { return font; }
+
+  std::shared_ptr<AssetManager> GetAssetManager() override { return nullptr; }
+
+  void OnRootIsolateCreated() override {};
+
+  void UpdateIsolateDescription(const std::string isolate_name,
+                                int64_t isolate_port) override {};
+
+  void SetNeedsReportTimings(bool value) override {};
+
+  std::unique_ptr<std::vector<std::string>> ComputePlatformResolvedLocale(
+      const std::vector<std::string>& supported_locale_data) override {
```

Comment:
Nothing to do in your patch, but I'm going to take a look at the base class. Really curious why we're handing back a `std::unique_ptr<std::vector<std::string>>` here rather than just a `std::vector<std::string>`. I doubt these arrays are ever more than 5-10 entries even in extreme cases.